plugin/trino-pulsar/src/main/java/io/trino/plugin/pulsar/PulsarSplitManager.java

```diff
@@ -0,0 +1,500 @@
+
+    private final PulsarConnectorConfig pulsarConnectorConfig;
+
+    private final PulsarConnectorCache pulsarConnectorManagedLedgerFactory;
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Inject
+    public PulsarSplitManager(
+            CatalogName catalogName,
+            PulsarConnectorConfig pulsarConnectorConfig,
+            PulsarConnectorCache pulsarConnectorManagedLedgerFactory)
+    {
+        this.catalogName = requireNonNull(catalogName, "catalogName is null").toString();
+        this.pulsarConnectorConfig = requireNonNull(pulsarConnectorConfig, "pulsarConnectorConfig is null");
+        this.pulsarConnectorManagedLedgerFactory = requireNonNull(pulsarConnectorManagedLedgerFactory, "pulsarConnectorManagedLedgerFactory is null");
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle transactionHandle,
+            ConnectorSession session,
+            ConnectorTableLayoutHandle layout,
+            SplitSchedulingStrategy splitSchedulingStrategy)
+    {
+        int numSplits = this.pulsarConnectorConfig.getTargetNumSplits();
+
+        PulsarTableLayoutHandle layoutHandle = (PulsarTableLayoutHandle) layout;
+        PulsarTableHandle tableHandle = layoutHandle.getTable();
+        TupleDomain<ColumnHandle> tupleDomain = layoutHandle.getTupleDomain();
+
+        String namespace = PulsarConnectorUtils.restoreNamespaceDelimiterIfNeeded(tableHandle.getSchemaName(), pulsarConnectorConfig);
+        TopicName topicName = TopicName.get("persistent", NamespaceName.get(namespace), tableHandle.getTopicName());
+
+        SchemaInfo schemaInfo;
+
+        try (PulsarAdmin pulsarAdmin = PulsarAdminClientProvider.getPulsarAdmin(pulsarConnectorConfig)) {
+            schemaInfo = pulsarAdmin.schemas().getSchemaInfo(format("%s/%s", namespace, tableHandle.getTopicName()));
+        }
+        catch (PulsarAdminException e) {
+            if (e.getStatusCode() == 401) {
+                throw new TrinoException(QUERY_REJECTED, format("fail to get pulsar topic schema for topic %s/%s: Unauthorized", namespace, tableHandle.getTopicName()));
+            }
+            else if (e.getStatusCode() == 404) {
+                schemaInfo = PulsarSchemaInfoProvider.defaultSchema();
+            }
+            else {
+                throw new TrinoException(PULSAR_SPLIT_ERROR, "fail to get pulsar topic schema", e);
+            }
+        }
+        catch (PulsarClientException e) {
+            throw new TrinoException(PULSAR_ADMIN_ERROR, "fail to create pulsar admin client", e);
+        }
+
+        Collection<PulsarSplit> splits;
+        try (PulsarAdmin pulsarAdmin = PulsarAdminClientProvider.getPulsarAdmin(pulsarConnectorConfig)) {
+            OffloadPoliciesImpl offloadPolicies = (OffloadPoliciesImpl) pulsarAdmin.namespaces()
+                                    .getOffloadPolicies(topicName.getNamespace());
+            if (offloadPolicies != null) {
```

Comment:
What if it is null? Is it ok to pass it to `getSplitsPartitionedTopic`? Maybe rnn here?